src/protocols/der/base.c

```diff
@@ -0,0 +1,518 @@
+/*
+ *   This library is free software; you can redistribute it and/or
+ *   modify it under the terms of the GNU Lesser General Public
+ *   License as published by the Free Software Foundation; either
+ *   version 2.1 of the License, or (at your option) any later version.
+ *
+ *   This library is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ *   Lesser General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Lesser General Public
+ *   License along with this library; if not, write to the Free Software
+ *   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/**
+ * $Id$
+ *
+ * @file protocols/der/decode.c
+ * @brief Functions to decode DER encoded data.
+ *
+ * @author Ethan Thompson (ethan.thompson@inkbridge.io)
+ *
+ * @copyright 2024 Inkbridge Networks SAS.
+ */
+
+#include "include/build.h"
```

Comment:
nits: we usually put `#include "foo"` after all of the common includes.  that way any location definitions don't affect the other header files.